### PR TITLE
refactor: Replace GitBook with MkDocs

### PR DESCRIPTION
**Pull request by x10an14**
https://www.mkdocs.org/

### CODE REVIEW

**Comment by x10an14** [comment]
Now rebased on top of current master (`fdf0205f81b89f89bfe376e654cae8e53b3fa1c3`).

**Comment by chinatsu** [comment]
Outsider peeking in here. I'm into this for a few reasons. 
1. I prefer this setup over GitBook because it feels as if GitBook can do [weird things in the repository it's connected to](https://github.com/nais/doc/commit/6237139118f18c7a0316fff3be6e012eeb57abaf). 
2. This setup is closer to the mechanism that generates pages, it's not some magical thing that happens over on GitBook's servers.
3. This is open source and extensible. Do you ever wish you wanted richer documentation? Chances are someone's already made an extension that you can plug in to cover your needs. If not, it's probably not a mountain of work to sort out something that parses your defined token and generate something nice. 
   * I can also imagine it's easy to plug in, say, Amplitude to see what pages people read most and consider if those should be closer to the "surface".
4. One can boot this up locally and see what changes look like in the real system with hot-reloading.
5. (Assuming one uses relative paths to point to internal resources everywhere) the compiler will throw warnings if links and image references don't point to existing resources.

I imagine some things can be a little problematic; mkdocs only places pages on leaf nodes, so there is no page for the category itself. There are a few (<20) places in the documentation so far that references categories, and some of the category pages do contain important information in the current version ([NAIS application](https://doc.nais.io/nais-application), [Deployment](https://doc.nais.io/deployment), and [naisdevice](https://doc.nais.io/device)). If the move to mkdocs is to be considered, I think some content needs to be restructured as to point directly to leaf nodes, be it an overview page or something else. 

I'll be thinking about how this could be sorted out on a content level, and I'm curious what thoughts you guys might have on this topic, so hit me up on Slack if you have some opinion!

**Comment by x10an14** [comment]
@chinatsu Thanks for your input! =D
> I'm into this for a few reasons.

Don't be so humble as to avoid mentioning that this was inspired by your work and efforts! Our gratitude to you my friend! =)

File: docs/addons/kafka.md [hunk 1]
```diff
@@ -43,8 +43,7 @@ Currently, use the `nav-dev` pool for development, and `nav-prod` for production
 | `nav-dev` | 2 | 3 | `dev-gcp` | `dev-gcp`, `dev-fss`, `dev-sbs` |
 | `nav-prod` | 2 | 3 | `prod-gcp` | `prod-gcp`, `prod-fss`, `prod-sbs` |
 
-{% tabs %}
-{% tab title="topic.yaml" %}
+
 ```yaml
```
Comment (Kyrremann) [inline docs/addons/kafka.md:47]: Ser ut som du bare har fjernet tabs her. 

```
=== ".github/workflows/deploy.yml"
    ```yaml
    name: Build, push, and deploy
```

**Comment by x10an14** [comment]
@nais/aura Any input on which GCP project the bucket would be fine to reside in? I think this needs to be decided before we set DNS, no?

Maybe even find a way for both sites to exist simultaneously for a short period.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,12 @@
+# IntelliJ / IDEA
 .idea
 *.iml
-.DS_Store
\ No newline at end of file
+
+# Mac OS shait
+.DS_Store
+
+# Vim
```
Comment (Kyrremann) [inline .gitignore:8]: Hvorfor heter ikke denne `Vim shait`? :p

File: docs/clusters/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Clusters
```
Comment (Kyrremann) [inline docs/clusters/README.md:1]: Når dette har blitt merget inn så bør vi se på å lage bedre hovedsider for hver kategori.

File: .gitignore (hunk 2 above)
Comment (x10an14) [reply .gitignore:8]: Fordi `# IntelliJ / IDEA` ikke har ` shait`suffiksen heller ;)

File: docs/clusters/README.md (hunk 3 above)
Comment (x10an14) [reply docs/clusters/README.md:1]: Ikke uenig, leker med å lage en egen mkdocs plugin som etteraper GitBook sin oppførsel (eller annen ønsket oppførsel) på fritiden.

**Comment by Kyrremann** [comment]
Vi har vel ikke noe nais-prosjekt, så da tenker jeg bucket hører inn under aura-prod.

**Review by Kyrremann — APPROVED** [review]
Da har jeg bladd meg igjennom alt, og ting ser riktig ut. Bra jobba!

Skulle vi ha merget den inn og manuelt brukt github pages til vi har fikset bucket?

**Comment by x10an14** [comment]
> @Kyrremann:
> Da har jeg bladd meg igjennom alt, og ting ser riktig ut. Bra jobba!
> 
> Skulle vi ha merget den inn og manuelt brukt github pages til vi har fikset bucket?

Usikker - bucket skulle igrunn vært fikset og klart nå, men noe har gått galt ([lenke her](https://newdoc.nais.io)). 
Sjekk `terraform` mappa jeg legger til PR branch nå.

Hvis du har kapasitet kan du sikkert ordne det slik, men vi må koordinere så vi ikke går i beina på hverandre mtp. DNS.

PS: Er også litt bekymret for (den minimale) risikoen å ha output av `mkdocs build` (les: HTMLen i `/site` lagt til i repo). 
Ligger ikke i `.gitignore` uten grunn^^

**Comment by x10an14** [comment]
Mangler også Workflow pipeline btw - det hadde jeg tenkt å ta imorra så snart bucket var klart.

**Pull request merged**